projects/plugins/jetpack/extensions/shared/external-media/media-service/index.ts

```diff
@@ -142,11 +143,17 @@ const buildMediaCategory = (
 			} ),
 			method: 'GET',
 		} )
-			.then( ( response: WpcomMediaResponse ) =>
-				response.media
+			.then( ( response: WpcomMediaResponse ) => {
+				const mediaItems = response.media
 					.filter( wpcomMediaItem => wpcomMediaItem.type === WpcomMediaItemType.Image )
-					.map( mapWpcomMediaToMedia )
-			)
+					.map( mapWpcomMediaToMedia );
+				jetpackAnalytics.tracks.recordEvent( 'jetpack_editor_media_inserter_external_source', {
+					mediaSource: source.toString(),
+					results: mediaItems.length,
+				} );
```

Comment:
It'd be interesting to also track the search:

```suggestion
				jetpackAnalytics.tracks.recordEvent( 'jetpack_editor_media_inserter_external_source', {
					mediaSource: source.toString(),
					results: mediaItems.length,
					search: mediaCategorySearch?.search === '' ? defaultSearch.search : mediaCategorySearch.search
				} );
```